src/node_constants.cc

```diff
@@ -795,11 +795,39 @@ void DefinePriorityConstants(Local<Object> target) {
 #endif
 }
 
+// Convert nid's to the string representation of their OID. Non-reentrant, and
+// will abort if called with invalid nids (so only pass values from OpenSSL's
+// headers).
+static const char* OBJ_nid2oid(int nid) {
+  ASN1_OBJECT* obj = OBJ_nid2obj(nid);
+  CHECK(obj);
+
+  static char buf[128];
+  CHECK_LE(OBJ_obj2txt(buf, sizeof(buf), obj, 1), 128);
```

Comment:
`sizeof(buf)`

Consider having the caller pass in the buffer: declare the param as `char (*buf)[128]`, the caller then passes in `&buf` and can't get the size wrong. (See `GetHumanReadableProcessName()` for another example.)

Reply:
I rewrote as you suggested, but I don't use `sizeof(buf)` because it is `size_t`, and `obj2txt()` returns `int`, and comparison causes warnings on sign mismatch. I can `static_cast<size_t>(OBJ_obj2txt())`, but that doesn't seem clearly better.